urllib3/util/url.py

```diff
@@ -192,7 +192,10 @@ def parse_url(url):
             # If given, ports must be integers.
             if not port.isdigit():
                 raise LocationParseError(url)
-            port = int(port)
+            try:
```

Comment:
If we're gonna put this in a try/except block, it probably is a bit redundant to keep the conditional above (lines 193-194), no?